### DISCUSSION TITLE
Problem Suggestion Thread

### DISCUSSION THREAD

**Discussion by SansPapyrus683** [cpinitiative/usaco-guide#3046] [Ideas]
decided to make a thread just for problem suggestions because of clutter
first gonna move all the previous suggestions here
add new suggestions in this thread i guess

**Comment by SansPapyrus683** [comment]
korean problems for the treap module:
* https://boj.kr/9548
* https://boj.kr/17777 <-- JOISC problem. It can also be easily solved using `__gnu_cxx::crope`

**Comment by SansPapyrus683** [comment]
range dp module:
https://hsin.hr/2010/school/seniors/tasks.pdf
judge link: https://boj.kr/3102

**Comment by SansPapyrus683** [comment]
korean problems for the counting mins w/ segtree module:
https://boj.kr/3392
https://boj.kr/2185
https://boj.kr/7626

**Comment by SansPapyrus683** [comment]
> I would like to propose this problem to be linked as an application (https://oj.uz/problem/view/JOI21_foodcourt). Although from what I heard the solution doesn't use beats per se (but a more relaxed version), it looks still as a good example

add in segtree beats?

**Comment by SansPapyrus683** [comment]
> https://atcoder.jp/contests/abc215/tasks/abc215_e can be added as a problem to this module.

in bitmask dp

**Comment by SansPapyrus683** [comment]
possible range dp problem: https://dmoj.ca/problem/pickit

**Comment by SansPapyrus683** [comment]
for binary jumping module:
> https://codeforces.com/problemset/problem/1184/E2
as a binary lifting/jumping problem (suggested difficulty: easy).
Note that this problem also requires knowledge on MST.
If it's decided to put this problem into the Guide, it's a good idea to note that reading
https://codeforces.com/contest/1184/problem/E1
before solving this problem is useful for understanding what the problem statement is asking.

**Comment by SansPapyrus683** [comment]
bitmask dp: https://codeforces.com/problemset/problem/1185/G1 (suggested difficulty: normal)

**Comment by SansPapyrus683** [comment]
sliding window module: https://codeforces.com/problemset/problem/1195/E

**Comment by Usernametaken12** [comment]
LIS module: up down subsequence: http://usaco.org/index.php?page=viewproblem2&cpid=1238
Probably normal difficulty-ish, definitely easier than Candy and the Global Warming, but harder than Cow Run and the xor problem.

**Reply by SansPapyrus683** [reply]
i feel like we've enough of these problems? unless you feel this contributes something really novel to the understanding

**Reply by Usernametaken12** [reply]
I feel like it takes some priority given that it is an usaco problem and that the module is lacking in that regard, having only has one other usaco problem-- one from 8 years ago. The gimmick of having multiple states of LIS is probably important to know to be able to incorporate it into harder problems (though I'm not sure how many problems actually use that). While that gimmick is somewhat shared by Global Warming, I don't think you'd instantly be able to solve one having solved the other especially due to the fact that global warming only works with prefixes/suffixes (contrary to my prior statement it's definitely easier than up-down, idk why global warming took me a similar time to solve compared to up-down), and is probably easier to most people than the other. I don't mind having problems that have a similar gimmick to make the problemset easier by giving multiple ways to get to a particular technique (see rectangles, fancy fence, and maximum building I in stacks). Besides, worst case people just skip the problems if they think of the solution immediately, or if they don't get it immediately the topic wasn't clearly wasn't completely learned and they could probably use another problem on it. 

Ultimately though I would agree that including this problem probably isn't the highest priority as the module is already very healthy and the topic doesn't occur particularly often in practice to warrant a ton of problems.

**Comment by SansPapyrus683** [comment]
BOI 2022 Passes for gold bitmask dp

**Comment by SansPapyrus683** [comment]
add CF 1472-F https://codeforces.com/contest/1472/problem/F to silver greedy

**Comment by SansPapyrus683** [comment]
> hi add this problem to Usaco gold Range DP section
https://codeforces.com/contest/1132/problem/F

**Reply by WilliamYuan42** [reply]
hi i stumbled upon this problem as i was grinding cf and i'm going to add a pull request for this soon:tm:
do you think it would fit better in 'easy' or 'normal'

**Reply by Usernametaken12** [reply]
I would say easy, personally, as this is very much a classical problem. Additionally, the transitions in this problem are very similar to the modern art problem in the module, and the problem is a touch easier due to being more explicit in the problem statement regarding the approach.

**Reply by WilliamYuan42** [reply]
ok, fair

**Reply by SansPapyrus683** [reply]
nah put it in normal imo

**Reply by Usernametaken12** [reply]
I'm not sure what makes this problem difficult. This problem is identical to Modern art 3 in all but the problem description. If this is bumped to normal so should modern art 3.

**Reply by SansPapyrus683** [reply]
lol idk who put modern art 3 in easy

**Reply by WilliamYuan42** [reply]
my face when ::::: ::: :: : : : : : : : : : : : : : : : :
i'll just leave it in easy for now

**Comment by SansPapyrus683** [comment]
https://oj.uz/problem/view/BOI21_prison for lagrangian relaxation or whatever that is

**Comment by SansPapyrus683** [comment]
time series tree on knapsack instead of dsu:
https://codeforces.com/contest/601/problem/E

**Comment by Usernametaken12** [comment]
"Harder" DSU 
https://tlx.toki.id/problems/apio-2020/B

**Comment by Usernametaken12** [comment]
Floyd-Warshall problem (first time I've ever seen one in the wild lol):

https://oj.uz/problem/view/APIO17_merchant 
Probably a bit hard for gold, but seeing how we only have one problem for it, it should probably go in.

**Comment by Sosuke23** [comment]
PURS Problem Suggestion:

Normal:
https://codeforces.com/contest/594/problem/D
https://codeforces.com/contest/501/problem/D
https://codeforces.com/contest/914/problem/D
https://codeforces.com/contest/301/problem/D
https://codeforces.com/contest/396/problem/C
https://codeforces.com/contest/61/problem/E

Hard:
https://codeforces.com/contest/19/problem/D
https://atcoder.jp/contests/abc343/tasks/abc343_f
